tests.py

```diff
@@ -1,81 +1,241 @@
-
 UI_OUTPUTS = [WELCOME_MESSAGE, INSTRUCTIONS, VALID_PATTERN, BYE_MESSAGE]
 
-# API TESTS
-"""
-def test_valid_login(url):
-    response = requests.post(url, data='text=great').json()
-    assert response.status_code == 200
-
-def test_valid_get_hint(url):
-    response = requests.get(url, data='text=great').json()
-    assert response.status_code == 200
-
-def test_valid_guess(url):
-    response = requests.put(url, data='text=great').json()
-    assert response.status_code == 200
-
-def test_valid_get_solution(url):
-    response = requests.get(url, data='text=great').json()
-    assert response.status_code == 200
-"""
-
 EMPTY_SUCCESS = "Hello and Welcome to the best hangman game ever. \n\nGame Over \n"
-DUMMY_SUCCESS = ""
-HTTPS_SUCCESS = ""
-SUCCESSES = [EMPTY_SUCCESS, DUMMY_SUCCESS, HTTPS_SUCCESS]
-
-
-def test_integration():
-    test_ui = TerminalUI()
-    test_methods = [EmptyMethod()]
-    # , DummyMethod(), HTTPSMethod()]
-    test_games = [Game(test_ui, test_method) for test_method in test_methods]
-
-    for test_index, test_game in enumerate(test_games):
-        proc = subprocess.Popen(['python', '/Users/kbenarie/Projects/hangman-exercise/main.py'],
-                                stdout=subprocess.PIPE)
-        stdout = proc.stdout.read().decode('utf-8')
-        assert stdout == SUCCESSES[test_index], '{} case did not succeed'.format(
-            SUCCESSES[test_index])
 
 
-def test_guess_letter():
-    letters = ['a', 'd', 't', 'g', 'r']
+# FUNCTIONS
+def validate_JSON(json_data):
+    """
+    This function validates json format on a given data structure
+    :param json_data:
+    :return: True if it is, False otherwise
+    """
+    try:
+        json.loads(json_data)
+    except ValueError as err:
+        return False
+    return True
+
+
+def edit_distance(string1, string2):
+    """
+    This function checks the length of the edit distance between two given strings
+    or in other words - how many operations should i do in order to get from string1 to string2.
+    Assumptions: The strings in this project are necessarily in the same length.
+    :param string1:
+    :param string2:
+    :return: the distance
+    """
+    count_diffs = 0
+    for a, b in zip(string1, string2):
+        if a != b:
+            count_diffs += 1
+    return count_diffs
+
+
+# CLASSES
+
+class TestClassCheckAPI(unittest.TestCase):
+    """
+    This test class was created in order to check the API that was given in this exercise.
+    """
+
+    def setUp(self):
+        """
+        the testing framework executes the test setup method first,
+        before any test method in the class. It includes all the required data for each test.
+        """
+        self.url = "https://hangman-api.herokuapp.com/hangman"
+        self.response = requests.post(self.url, data='text=great').json()
+        self.game_token = self.response['token']
+        self.hangman_string = self.response['hangman']
+
+    def test_types(self):
```

Comment:
What exactly are we testing here? What needs to be broken in order for this test to fail?

Reply:
we spoke about it. i agree.